### PR TITLE
hyprls: init at 0.2.0

### PR DESCRIPTION
**Pull request by arthsmn**
## Description of changes
This is a hyprlang LSP. Currently using it in helix just fine.

Homepage: https://en.ewen.works/hyprls
Source code: https://github.com/hyprland-community/hyprls

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/hy/hyprls/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  vendorHash = "sha256-rG+oGJOABA9ee5nIpC5/U0mMsPhwvVtQvJBlQWfxi5Y=";
+
+  checkFlags = [
+    # Not yet implemented
+    "-skip=TestHighLevelParse"
+  ];
+
+  meta = with lib; {
+    description = "A LSP server for Hyprland's configuration language";
+    homepage = "https://en.ewen.works/hyprls";
```
Comment (getchoo) [inline pkgs/by-name/hy/hyprls/package.nix:25]:
```suggestion
    homepage = "https://en.ewen.works/hyprls";
    changelog = "https://github.com/hyprland-community/hyprls/releases/tag/v${version}";
```

File: pkgs/by-name/hy/hyprls/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    license = licenses.mit;
+    maintainers = with maintainers; [ arthsmn ];
+    platforms = platforms.all;
```
Comment (getchoo) [inline pkgs/by-name/hy/hyprls/package.nix:28]:
```suggestion
```
`buildGoModule` handles this already

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/hy/hyprls/package.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  vendorHash = "sha256-rG+oGJOABA9ee5nIpC5/U0mMsPhwvVtQvJBlQWfxi5Y=";
+
+  checkFlags = [
+    # Not yet implemented
+    "-skip=TestHighLevelParse"
+  ];
+
+  meta = with lib; {
+    description = "A LSP server for Hyprland's configuration language";
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hyprls/package.nix:24]:
```suggestion
    description = "LSP server for Hyprland's configuration language";
```

File: pkgs/by-name/hy/hyprls/package.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  vendorHash = "sha256-rG+oGJOABA9ee5nIpC5/U0mMsPhwvVtQvJBlQWfxi5Y=";
+
+  checkFlags = [
+    # Not yet implemented
+    "-skip=TestHighLevelParse"
+  ];
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hyprls/package.nix:23]: Better to avoid `with lib`

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/hy/hyprls/package.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    hash = "sha256-boA2kWlHm9bEM/o0xi/1FlH6WGU4wL1RRvbGGXdzHYQ=";
+  };
+
+  vendorHash = "sha256-rG+oGJOABA9ee5nIpC5/U0mMsPhwvVtQvJBlQWfxi5Y=";
+
+  checkFlags = [
+    # Not yet implemented
+    "-skip=TestHighLevelParse"
+  ];
+
```
Comment (getchoo) [inline pkgs/by-name/hy/hyprls/package.nix:22]:
```suggestion

  passthru = {
    updateScript = nix-update-script { };
  };

```
should help make this easier to maintain

File: pkgs/by-name/hy/hyprls/package.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
```
Comment (getchoo) [inline pkgs/by-name/hy/hyprls/package.nix:3]:
```suggestion
, fetchFromGitHub
, nix-update-script
```

File: pkgs/by-name/hy/hyprls/package.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    # Not yet implemented
+    "-skip=TestHighLevelParse"
+  ];
+
+  meta = {
+    description = "LSP server for Hyprland's configuration language";
+    homepage = "https://en.ewen.works/hyprls";
+    changelog = "https://github.com/hyprland-community/hyprls/releases/tag/v${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ arthsmn ];
```
Comment (getchoo) [inline pkgs/by-name/hy/hyprls/package.nix:28]:
```suggestion
    maintainers = with lib.maintainers; [ arthsmn ];
    mainProgram = "hyprls";
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Review by getchoo — APPROVED** [review]

**Comment by ewen-lbh** [comment]
Thanks a lot! Will definitely mention Nix support in the readme ^^

**Pull request merged**